service/history/workflow/delete_manager.go

```diff
@@ -95,21 +95,24 @@ func (m *DeleteManagerImpl) AddDeleteWorkflowExecutionTask(
 	visibilityQueueAckLevel int64,
 ) error {
 
-	// Create `DeleteWorkflowExecutionTask` only if workflow is closed successfully and all pending tasks are completed (if in active cluster).
-	// Otherwise, mutable state might be deleted before close tasks are executed (race condition between close and delete tasks).
+	// In active cluster, create `DeleteWorkflowExecutionTask` only if workflow is closed successfully
+	// and all pending transfer tasks are completed.
+	// This check is required to avoid race condition between close and delete tasks.
+	// Otherwise, mutable state might be deleted before close transfer task is executed, and therefore close task will be dropped.
+	//
+	// In passive cluster, transfer task queue check can be ignored but not visibility task queue.
+	// If visibility close task is executed after visibility is deleted then it will resurrect visibility record in closed state.
```

Comment:
The visibility request version (taskID) can't help in this case? because the record is deleted?

Reply:
yes, record is deleted, nothing to compare with. Also standard visibility doesn't have version check.